src/vs/workbench/services/search/node/rawSearchService.ts

```diff
@@ -344,3 +366,49 @@ interface CacheStats {
 	cacheFilterStartTime: number;
 	cacheFilterResultCount: number;
 }
+
+/**
+ * Collects a batch of items that each have a size. When the cumulative size of the batch reaches 'maxBatchSize', it calls the callback.
+ * If the batch isn't filled within 'timeout' ms, the callback is also called.
+ * And after 'batchOnlyAfter' items, the timeout is ignored, and the callback is called only when the batch is full.
```

Comment:
Yeah I mean it as "after N, only batch" but it also looks like "only batch after N, not before". Maybe `stopTimeoutAfter` or `timeoutItemsLimit` or something...?